### PR TITLE
Patch 2

### PR DESCRIPTION
**Pull request by jmonbillard**

### CODE REVIEW

File: bin/spotxBidAdapter.js [hunk 1]
```diff
@@ -305,56 +316,7 @@ export const spec = {
 
 function outstreamRender(bid) {
   if (bid.renderer.config.outstreamFunction != null && typeof bid.renderer.config.outstreamFunction === 'function') {
-    bid.renderer.config.outstreamFunction();
-    return;
-  }
-
-  window.console.log('[SPOTX][renderer] Handle SpotX outstream/inbanner renderer');
```
Comment (strider820) [inline bin/spotxBidAdapter.js:300]: why did you remove this whole renderer section?

File: bin/spotxBidAdapter.js [hunk 2]
```diff
@@ -244,7 +254,8 @@ export const spec = {
                 player_height: request.video.ext.player_height,
                 content_page_url: request.video.ext.content_page_url,
                 outstream_static: request.video.ext.outstream_static,
-                ad_mute: request.video.ext.ad_mute
+                ad_mute: request.video.ext.ad_mute,
+                outstreamFunction: request.video.ext.outstreamFunction
```
Comment (strider820) [inline bin/spotxBidAdapter.js:245]: what is this?

File: bin/spotxBidAdapter.js (hunk 1 above)
Comment (jmonbillard) [reply bin/spotxBidAdapter.js:300]: Because it's smarter to keep that externally to the adaptor code in order to have different rendering possibilities by publisher. Thus, the idea is to share a sample prebid config file that is doing a classic rendering, and depending on their needs we can adapt the code client side without having to handle rendering option in a static adaptor code. Sorry if f I am not clear, you can check the implementation here http://testweb.spotx.tv/jmonbillard/spotx-prebid-goodAdaptor-outstream.html

File: bin/spotxBidAdapter.js (hunk 2 above)
Comment (jmonbillard) [reply bin/spotxBidAdapter.js:245]: It's a reference to the rendering function created client side with the prebid json config.

I am setting up the outstream function using renderer.setRender(outstreamRender);

AND then I am calling the client side rendering function in this function when renderAd gets called.

Let me know if you have questions. You can observe the behavior in my testing page.

**Review by jmonbillard — COMMENTED** [review]
Last version of the adaptor. Working on my testings env. Including your changes and my last changes.

**Comment by jmonbillard** [comment]
http://testweb.spotx.tv/jmonbillard/spotx-prebid-goodAdaptor-outstream.html

File: bin/spotxBidAdapter.js (hunk 1 above)
Comment (strider820) [reply bin/spotxBidAdapter.js:300]: That's true, but we should leave this code in here, for pubs that don't want to have their own renderer. The way that prebid works, is if the publisher has registered a renderer on their side, then prebid will use that. But if that doesn't exist, it would use this

Comment (strider820) [reply bin/spotxBidAdapter.js:300]: http://prebid.org/dev-docs/show-outstream-video-ads.html#renderers

File: bin/spotxBidAdapter.js (hunk 2 above)
Comment (strider820) [reply bin/spotxBidAdapter.js:245]: If I'm understanding this correctly, I think my answer above applies here too. If the pub has a renderer, and we want to help them with that, that's fine, but we should assume that's not the case. I'm fine with leaving this option in here, and short circuiting the renderer that we had, but we should still support the pub not doing this, and not having a renderer.

**Review by jmonbillard — COMMENTED** [review]
last changes

File: bin/spotxBidAdapter.js (hunk 1 above)
Comment (jmonbillard) [reply bin/spotxBidAdapter.js:300]: I just pushed a new version including default renderer so publishers does not have to add it on their page. They can still choose to override the default one by a more custom one.

File: bin/spotxBidAdapter.js (hunk 2 above)
Comment (jmonbillard) [reply bin/spotxBidAdapter.js:245]: Same as above

**Review by strider820 — APPROVED** [review]
